connection.go

```diff
@@ -1474,7 +1439,7 @@ func (conn *Connection) NewWatcher(key string, callback WatchCallback) (Watcher,
 	// That's why we can't just check the Tarantool response for an unsupported
 	// request error.
 	if !isFeatureInSlice(iproto.IPROTO_FEATURE_WATCHERS,
-		conn.opts.RequiredProtocolInfo.Features) {
+		conn.serverProtocolInfo.Features) {
```

Comment:
This is not the same as before:

Previously, we required this from the server, now we check availability.

I think it's not a big deal, but we need to add the change into the changelog.

Reply:
I return `RequiredProtocolInfo` back to `DialOpts` (and, also in `Opts`) , in order not to write crutches to find dialer/conn, because we need check it also [here](https://github.com/tarantool/go-tarantool/blob/master/pool/connection_pool.go#L933).

Reply:
It could be better to check the server protocol info instead of required because we don't have such checks from the client requirements to enable other features.